raiden/api/python.py

```diff
@@ -864,20 +871,41 @@ def get_token_network_events_raiden(
             to_block=to_block,
         )
 
-        for block_number, event in raiden_events:
-            new_event = {
-                'block_number': block_number,
-                'event': type(event).__name__,
-            }
-            if hasattr(event, 'transfer'):
-                if event.transfer.token == token_address:
-                    event.transfer = repr(event.transfer)
-            elif hasattr(event, 'token'):
-                if event.token == token_address:
-                    event.balance_proof = repr(event.balance_proof)
-            new_event.update(event.__dict__)
-            returned_events.append(new_event)
-        returned_events.sort(key=lambda evt: evt.get('block_number'), reverse=True)
+        returned_events = [
+            (block_number, event)
+            for block_number, event in raiden_events
+            if self._is_internal_event(event, token_address)
+        ]
+
+        returned_events.sort(key=lambda event: event[0], reverse=True)
         return returned_events
 
+    def _is_internal_event(
+            self,
+            event,
+            token_address,
+    ):
+        if hasattr(event, 'transfer') and event.transfer.token == token_address:
```

Comment:
I believe this is a bad approach. If you expect an event or a set of events to have a transfer, then check `isinstance(event, 'ThatSpecificEvent')` or `isinstance(event, 'CommonParentClassOfThatEvent')`.

Reply:
I thought about doing that but the if statements will get longer. Mainly because there are events which I would have to check. This is more efficient, since for example SendLockedTransfer, SendBalanceProof, SendRefundTransfer all of them have a recipient. 

Also this endpoint might get removed. It doesn't have to be perfect. Thanks for the feedback @rakanalh , I will do that from now on.